### PR TITLE
NF: Adds SurfaceRingQueryEngine for surfaces similar to HollowSphere

### PR DESCRIPTION
**Pull request by swaroopgj**
Adds SurfaceRingQueryEngine for surfaces similar to HollowSphere,
with tests and some PEP8 formatting for test_surfing.
@nno @yarikoptic

### CODE REVIEW

**Comment by yarikoptic** [comment]
Hey @nno -- could you remind me why I haven't insisted on having similar to volumetric QueryEngines to have clean separation of "geometry" (Sphere, HollowSphere) from the QE itself?

**Comment by nno** [comment]
@yarikoptic I don't know why you did not insist on that. Possibly because "geometry" objects for surfaces have to contain the Surface objects themselves anyway. Alternatively, with only one geometry at the time (standard circle on the surface) this would have seemed overkill back then.

File: mvpa2/misc/surfing/queryengine.py [hunk 1]
```diff
@@ -188,6 +188,111 @@ def query_byid(self, vertex_id):
...
+        v2f = self._vertex2feature_map
+        # Sorting nodes based on distance to center node to work around
+        # the problem with add_center_fa in Searchlight
+        nearby_nodes_keys = sorted(nearby_nodes, key=nearby_nodes.__getitem__)
+        if self.include_center and vertex_id in nearby_nodes_keys:
+            return sum((v2f[node] for node in nearby_nodes_keys
+                        if nearby_nodes[node] > self.inner_radius), v2f[vertex_id])
+        else:
+            return sum((v2f[node] for node in nearby_nodes
+                        if nearby_nodes[node] > self.inner_radius), [])
```
Comment (nno) [inline mvpa2/misc/surfing/queryengine.py]: there seems some code duplication here for lines 290 and 293.

**Comment by nno** [comment]
@swaroopgj nice work. Would your approach to sorting the nodes as done here also fix issue #447?

**Comment by swaroopgj** [comment]
Not directly, it could still provide wrong ROI_seed, but atleast we can be
sure that the first node in the list is the center or the closest to the
center.

**Comment by nno** [comment]
@swaroopgj looks good. @yarikoptic any objections to merging?

**Pull request merged**